src/validator/is-ipv6-addr.ts

```diff
@@ -0,0 +1,74 @@
+import {STRule} from '../rule/rule';
+import {STRuleFn} from '../rule/fn';
+import {STRuleModifiers} from '../rule/modifiers';
+import {STRuleNode} from '../rule/node';
+import {STRuleNodeType} from '../rule/node-type';
+
+export type STOpIsIpv6Addr<CallerType> = () => CallerType;
+
+export const isIpv6Addr = (current: string): boolean => {
+	if (typeof current != 'string') {
+		return false;
+	}
+
+	const trimmed = current.trim();
+
+	if (!trimmed) {
+		return false;
+	}
+
+	const section = trimmed.split(':');
+	const doubleColonSegment = current.split('::');
+	const doubleColon = '::';
+
+	if (!section.length) {
+		return false;
+	}
+
+	if (section.length >= 9) {
+		return false;
+	}
+
+	if (section.length === 8 && !section.every(isValidSegment) && current.includes(doubleColon)) {
```

Comment:
Small performance adjustment here. Consider that `section.every` is an `O(n)` operation where `n` is the number of array elements. Here we're working with small numbers since the segment count is limited to 8, but between line 32 and line 36 we're doubling the number of operations needed by using every twice.

Move the every segment validation check above line 32 and use the resuult in both the `section.length` 8 and `section.length` 7 tests, and do an early drop and you can remove it from the length checks entirely.

Both checks bail out if all segments are not valid. And they use the same array input, so we can move it up and bail out if its false.
```typescript
const allValid = section.every(isValidSegment);
if (!allValid) {
   return false;
}


if (section.length === 8 && current.includes(doubleColon)) {
...
```